deltaspike/modules/data/impl/src/main/java/org/apache/deltaspike/data/impl/builder/AnnotatedQueryBuilder.java

```diff
@@ -88,4 +83,12 @@ else if (query.isNative())
         return context.applyRestrictions(result);
     }
 
+    private Class<?> getMethodResultEntityClass(Method m){
```

Comment:
HI Danny, good catch, thats what happens when I'm looking at this too late in the evening.

Can you please commit this all under the original ticket?  and can you expand this method to check for collection types?